### PR TITLE
[FIX] Directory sort and column sizes were wrong

### LINKED ISSUE #9817: Discovery sort ui doesnt update
### Description:

discovery-sort

Clicking sort in the sort doesn't update the UI to reflect the sort order.

Also does not seem to work properly?  This is on community server, I know for a fact there are rooms that should have showed up when flipping user count.  Like general/support

### Server Setup Information:

* Version of Rocket.Chat Server: 0.62.0-rc.0

### LINKED ISSUE #10024: Directory channel listing has incorrect behaviour in high Y resolutions
### Description:
Directory does not list all channels when placing Rocket.Chat on a monitor in portrait mode (resolution based). 

### Server Setup Information:
* Version of Rocket.Chat Server: 0.62.1
* Operating System: Ubuntu 16.04.3 LTS
* Deployment Method(snap/docker/tar/etc): Snap
* Number of Running Instances: 1
* DB Replicaset Oplog: Enabled
* Node Version: v8.8.1
* mongoDB Version: ?

### Steps to Reproduce:

1. Place Chrome tab in fullscreen on a monitor in portrait mode (with height more than ~1200 pixels)
2. Click directory button

### Expected behavior:
All the channels available should be listed

### Actual behavior:
Few channels are listed, and cannot scroll further. Sorting by name "works" since the end of the channel list appears at the top, but scrolling is still not possible.

### Additional information
Different resolutions changes the behaviour:
1920x1200 (landscape) Works
1200x1920 (portrait) Doesn't work

Resolutions tested on the same monitor with same orientation:
2560x1440 (landscape) Doesn't work
1920x1080 (landscape) Works

So it would seem that the resolution is the culprit in this case,.

### PR DESCRIPTION
**Pull request by ggazzo**
we can`t  sort by array length, (actually we can but its very expensive)

closes #9817
closes #10024
before: 
image
 after:
image

### CODE REVIEW

File: packages/rocketchat-theme/client/imports/components/modal/directory.css [hunk 1]
```diff
@@ -34,28 +36,40 @@
...
 		display: flex;
-		margin: 0 9px;
+
 		flex-direction: column;
-		width: 100%;
+		width: 1%;
+		flex-grow: 1;
+
+		white-space: nowrap;
+    overflow: hidden;
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/components/modal/directory.css]: wrong indentation

File: packages/rocketchat-ui/client/views/app/directory.html [hunk 2]
```diff
@@ -19,12 +19,12 @@
...
 		<div class="rc-directory-content">
 			{{#if $eq searchType 'channels'}}
-				<table class="rc-table">
+				<table class="rc-table rc-table--fixed">
 					<thead class="rc-table-head">
 						<tr class="rc-table-tr">
 							<td class="rc-table-td js-sort" data-sort="name">Name {{> icon icon=(sortIcon 'name')}}</td>
-							<td class="rc-table-td js-sort" data-sort="usernames">Users {{> icon icon=(sortIcon 'usernames') }}</td>
-							<td class="rc-table-td js-sort" data-sort="createdAt">Created At {{> icon icon=(sortIcon 'createdAt') }}</td>
+							<td class="rc-table-td rc-table-td--users">Users</td>
```
Comment (graywolf336) [inline packages/rocketchat-ui/client/views/app/directory.html]: The `Users` is not i18n'd.

File: packages/rocketchat-ui/client/views/app/directory.html [hunk 3]
```diff
@@ -19,12 +19,12 @@
 		{{/header}}
 		<div class="rc-directory-content">
 			{{#if $eq searchType 'channels'}}
-				<table class="rc-table">
+				<table class="rc-table rc-table--fixed">
 					<thead class="rc-table-head">
 						<tr class="rc-table-tr">
 							<td class="rc-table-td js-sort" data-sort="name">Name {{> icon icon=(sortIcon 'name')}}</td>
```
Comment (graywolf336) [inline packages/rocketchat-ui/client/views/app/directory.html]: The `Name` is not i18n'd.

File: packages/rocketchat-ui/client/views/app/directory.html [hunk 4]
```diff
@@ -19,12 +19,12 @@
...
-							<td class="rc-table-td js-sort" data-sort="usernames">Users {{> icon icon=(sortIcon 'usernames') }}</td>
-							<td class="rc-table-td js-sort" data-sort="createdAt">Created At {{> icon icon=(sortIcon 'createdAt') }}</td>
+							<td class="rc-table-td rc-table-td--users">Users</td>
+							<td class="rc-table-td js-sort rc-table-td--createdAt" data-sort="createdAt">Created At {{> icon icon=(sortIcon 'createdAt') }}</td>
```
Comment (graywolf336) [inline packages/rocketchat-ui/client/views/app/directory.html]: The `Created At` is not i18n'd.

**Comment by karlprieb** [comment]
@graywolf336 done :) Thank you for your review 👍

**Pull request merged**